rust/src/dns/detect.rs

```diff
@@ -26,6 +26,12 @@ pub struct DetectDnsOpcode {
     opcode: u8,
 }
 
+#[derive(Debug, PartialEq, Eq)]
+pub struct DetectDnsRcode {
```

Comment:
We should reuse `DetectUintData<u8>` cf https://github.com/OISF/suricata/pull/10087#discussion_r1434504195

Reply:
It was suggested here https://github.com/OISF/suricata/pull/10087#issuecomment-1867871153 that I can use simple `u8` till the code is in working condition. I can change it to `DetectUintData<u8>` in the next PR I create.